pkgs/tools/security/sbomnix/default.nix

```diff
@@ -1,53 +1,209 @@
-{ lib
-, fetchFromGitHub
-, coreutils
-, curl
-, gnugrep
-, gnused
-, gzip
-, nix
-, python
+{
+  lib,
+  fetchFromGitHub,
+  grype,
+  nix,
+  python,
+  vulnix,
   # python libs
-, colorlog
-, graphviz
-, numpy
-, packageurl-python
-, pandas
-, requests
-, reuse
-, tabulate
+  beautifulsoup4,
+  binaryornot,
+  boolean-py,
+  colorlog,
+  dataproperty,
+  debian,
+  graphviz,
+  jinja2,
+  license-expression,
+  matplotlib,
+  mbstrdecoder,
+  networkx,
+  numpy,
+  packageurl-python,
+  packaging,
+  pandas,
+  pathvalidate,
+  poetry-core,
+  pygraphviz,
+  pyrate-limiter,
+  requests,
+  requests-cache,
+  requests-ratelimiter,
+  reuse,
+  tabledata,
+  tabulate,
+  typepy,
+  typing-extensions
 }:
-
 python.pkgs.buildPythonApplication rec {
   pname = "sbomnix";
-  version = "1.4.5";
+  version = "1.6.1";
 
   src = fetchFromGitHub {
     owner = "tiiuae";
-    repo = pname;
+    repo = "sbomnix";
     rev = "refs/tags/v${version}";
-    hash = "sha256-RxDFxVGivVBw2uhtzf231Q2HHTBFKSqGrknr2Es/ygM=";
+    hash = "sha256-kPjCK9NEs3D0qFsSSVX6MYGKbwqeij0svTfzz5JC4qM=";
+
+    # Remove documentation as it contains references to nix store
+    postFetch = ''
+      rm -fr "$out"/doc
+      find "$out" -name '*.md' ! -name "README.md" -exec rm -f '{}' \;
+    '';
   };
 
-  makeWrapperArgs = [
-    "--prefix PATH : ${lib.makeBinPath [ coreutils curl gnugrep gnused gzip graphviz nix ]}"
-  ];
+  postInstall = ''
+
+    wrapProgram $out/bin/sbomnix \
+        --prefix PATH : ${lib.makeBinPath [nix graphviz]}
+
+    wrapProgram $out/bin/nixgraph \
+        --prefix PATH : ${lib.makeBinPath [nix graphviz]}
+
+    wrapProgram $out/bin/vulnxscan \
+        --prefix PATH : ${lib.makeBinPath [grype nix vulnix-patched]}
+
+    wrapProgram $out/bin/nix_outdated \
+        --prefix PATH : ${lib.makeBinPath [nix-visualize]}
+
+    wrapProgram $out/bin/provenance \
+        --prefix PATH : ${lib.makeBinPath [nix]}
+
+  '';
+
+  # We need to patch vulnix to add support for runtime-only scan
+  # ('-C' command-line option) which is currently not available in
+  # released version of vulnix.
+  # Pending: https://github.com/nix-community/vulnix/pull/80.
+  vulnix-patched = vulnix.overrideAttrs (
+    old: rec {
+      patches = (old.patches or [ ]) ++ [
+        ./patches/vulnix/0001-remove-unused-function-parameter.patch
+        ./patches/vulnix/0002-add-flag-to-scan-only-runtime-dependencies-store-out.patch
+        ./patches/vulnix/0003-rename-runtime-to-closure.patch
+        ./patches/vulnix/0004-add-shorthand-C-for-closure.patch
+        ./patches/vulnix/0005-improve-comment.patch
+        ./patches/vulnix/0006-fix-failure-to-find-deriver-for-derivation-input-sou.patch
+        ./patches/vulnix/0007-avoid-unnecessary-nix-invocation-during-closure-scan.patch
+      ];
+    }
+  );
+
+  # https://github.com/craigmbooth/nix-visualize (not available in nixpkgs)
+  nix-visualize = python.pkgs.buildPythonApplication {
```

Comment:
This is nixpkgs, so it needs to be added to the python package set (and needs `meta`).

Reply:
Fixed in https://github.com/NixOS/nixpkgs/pull/299429/commits/102642cbcf42b05c90690e624f606bbc24ede286